core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/RMInterface.java

```diff
@@ -137,6 +163,130 @@ public RMInterface() throws IOException {
 
     }
 
+    public void pushRegionEpoch (HTableDescriptor desc, final TransactionState ts) throws IOException {
+       LOG.info("pushRegionEpoch start; transId: " + ts.getTransactionId());
+
+       TransactionalTable ttable1 = new TransactionalTable(Bytes.toBytes(desc.getNameAsString()));
+       HConnection connection = ttable1.getConnection();
+       long lvTransid = ts.getTransactionId();
+       RegionLocator rl = connection.getRegionLocator(desc.getTableName());
+       List<HRegionLocation> regionList = rl.getAllRegionLocations();
+
+       boolean complete = false;
+       int loopCount = 0;
+       int result = 0;
+
+       for (HRegionLocation location : regionList) {
+          final byte[] regionName = location.getRegionInfo().getRegionName();
+          if (compPool == null){
+              LOG.info("pushRegionEpoch compPool is null");
+              threadPool = Executors.newFixedThreadPool(intThreads);
+              compPool = new ExecutorCompletionService<Integer>(threadPool);
+          }
+
+          final HRegionLocation lv_location = location;
+          final HConnection lv_connection = connection;
+          compPool.submit(new RMCallable2(ts, lv_location, lv_connection ) {
+             public Integer call() throws IOException {
+                return pushRegionEpochX(ts, lv_location, lv_connection);
+             }
+          });
+          try {
+            result = compPool.take().get();
+          } catch(Exception ex) {
```

Comment:
As part of Exception handling, there has been an effort to catch specific exceptions only. Can you please modify this code accordingly.